src/microbe_stage/systems/OsmoregulationAndHealingSystem.cs

```diff
@@ -86,7 +86,13 @@ protected override void Update(float delta, in Entity entity)
             if (compounds.TakeCompound(atp, cost) < cost)
             {
                 // Ran out of ATP, disable engulf
-                control.SetStateColonyAware(entity, MicrobeState.Normal);
+                // control.SetStateColonyAware(entity, MicrobeState.Normal);
+
+                health.CurrentHealth -= Constants.ENGULF_NO_ATP_DAMAGE;
```

Comment:
Doesn't this cause damage really fast as this runs 60 times per second?

I'd prefer an approach where the damage was applied when applying the effect. And there was a new property like `control.ForcedStateDuration` that protects getting kicked out of engulf mode while it lasts.

That way the AI doesn't need changes to avoid killing itself accidentally, and the damage can be applied all at once and the message is sent just once when the player enables the engulf mode.

If this gradual damage approach is kept the damage *must be* multiplied by delta here as otherwise the damage amount is inconsistent if the player doesn't get a steady 60 FPS.

Reply:
I think I lean towards having it in the engulfing system, that way it is logically in the right place (engulfing energy cost in engulfing system rather than in *osmoregulation and healing* system).